### PR TITLE
[occm] Added Octavia Availability Zones support for LoadBalancers

### LINKED ISSUE #1223: [openstack-cloud-controller-manager] Support Octavia Availability Zone
/kind feature

**What happened**:
The Ussuri release of Octavia now supports the availability zone feature where operators can specify some useful capabilities such as the compute availability zone for the amphora instances. For large distributed clouds (enterprise and public), this is a must-have feature that not only improves reliability but also allows operators to use their hardware efficiently. Effectively and according to the tasks. Currently, we have to create a separate zone only for load balancers (with one huge lb-mng network) and all cloud projects use such a hardware pool which sometimes may experience conflicts and resources overflow. This also leads to one point of failure.

**What you expected to happen**:
I want to suggest to add the "availability-zone" parameters which will be used in Octavia LB request [1]

```
{
    "loadbalancer": {
        "name": "best_load_balancer",
        "provider": "amphora",
        "availability_zone": "my_az",
        "vip_subnet_id": "d4af86e1-0051-488c-b7a0-527f97490c9a",
    }
}
```

The same logic as for flavor-id: [2]

It looks like gophercloud doesnt have such an attribute [3] [4]

[1] https://docs.openstack.org/api-ref/load-balancer/v2/?expanded=create-a-load-balancer-detail#create-a-load-balancer
[2] https://github.com/kubernetes/cloud-provider-openstack/blob/master/pkg/cloudprovider/providers/openstack/openstack_loadbalancer.go#L474
[3] https://github.com/gophercloud/gophercloud/blob/master/openstack/loadbalancer/v2/loadbalancers/requests.go
[4] https://github.com/gophercloud/gophercloud/issues/2021 
**Environment**:
- openstack-cloud-controller-manager(or other related binary) version: 1.19
- OpenStack version: Ussuri

### PR DESCRIPTION
**Pull request by ITD27M01**
**What this PR does / why we need it**:
The Ussuri release of Octavia now supports the availability zone feature where operators can specify some useful capabilities such as the compute availability zone for the amphora instances. 

A POST request to create LB:
https://github.com/openstack/octavia/blob/master/octavia/api/v2/types/load_balancer.py#L109-L126

Added in the 2.14 Octavia version:
https://github.com/openstack/octavia/blob/master/octavia/api/root_controller.py#L109

Such support in gophercloud: https://github.com/gophercloud/gophercloud/pull/2026
**Which issue this PR fixes(if applicable)**:
fixes #1223

**Special notes for reviewers**:

The code to check the availability of the Octavia feature will inform operators through the warning if Octavia doesn't support AZ yet, but the operator requests it (in the cloud config or service annotation):

```
I0924 08:41:21.765794       1 controller.go:368] Ensuring load balancer for service kube-system/prometheus-operator-grafana-lb
W0924 08:41:21.766243       1 openstack_loadbalancer.go:1368] LoadBalancer Availability Zones isn't supported. Upgrade Octavia API to version 2.14 (Ussuri release) or later to use them
I0924 08:41:21.768166       1 event.go:291] "Event occurred" object="kube-system/prometheus-operator-grafana-lb" kind="Service" apiVersion="v1" type="Normal" reason="EnsuringLoadBalancer" message="Ensuring load balancer"
```

```release-note
[openstack-cloud-controller-manager] Added Octavia Availability Zones support for LoadBalancers.
```

### CODE REVIEW

**Comment by ITD27M01** [comment]
/assign @lingxiankong
Hello, I've tested this for Train and Ussuri

**Comment by jichenjc** [comment]
/lgtm

seems the error is not related
not expert in LB related topics but the change looks reasonable

File: docs/using-openstack-cloud-controller-manager.md [hunk 1]
```diff
@@ -204,6 +204,9 @@ Although the openstack-cloud-controller-manager was initially implemented with N
 * `flavor-id`
   The id of the loadbalancer flavor to use. Uses octavia default if not set.
 
+* `availability-zone`
+  The name of the loadbalancer availability zone to use. Applicable if use-octavia is set to True and requires Octavia API version 2.14 or later (Ussuri release). The Octavia availability zone capabilities will not be used if the parameter is not set.
```
[the remark this answers is not in the document]
Comment (ITD27M01) [reply docs/using-openstack-cloud-controller-manager.md]: I updated the description. In general, the behavior is the same as for the above mentioned flavor-id.

[the remark this answers is not in the document]
Comment (lingxiankong) [reply docs/using-openstack-cloud-controller-manager.md]: Thanks, we were not very strict for the config options and annotations added before, some of the descriptions are not user friendly. We need to make them more clear for how to properly use those options/annotations.

**Comment by lingxiankong** [comment]
/override openlab/cloud-provider-openstack-acceptance-test-csi-manila

**Pull request merged**